### PR TITLE
Plugins: Support templated urls in plugin routes

### LINKED ISSUE #16835: Support templated urls in plugin routes
**What would you like to be added**:
Support for templated urls in app plugin routes in the similar manner as templated urls in [datasource plugin routes/authentication](https://grafana.com/docs/plugins/developing/auth-for-datasources/#how-token-authentication-works) are currently supported.

**Why is this needed**:
An app that wants to support multi tenancy an inject the tenant dynamically in a url in the app backend proxy can't implement this today in a secure way since it currently must be solved in frontend where the url can be tampered.

### PR DESCRIPTION
**Pull request by briangann**
**What this PR does / why we need it**:

app plugins do not currently support templated urls in routes.  this lets the app use dynamic urls.

**Which issue(s) this PR fixes**:
Closes #16835

**Special notes for your reviewer**:

* Added code to implement interpolation of route urls using jsondata/securejsondata
* Extracted interpolateString() being used across plugin proxy code to a utils file
* Added tests for dynamic urls (variable interpolation for app plugins using pluginproxy)
* fixed meta-linter warnings (capitalization)

### CODE REVIEW

File: pkg/api/pluginproxy/pluginproxy.go [hunk 1]
```diff
@@ -38,6 +39,29 @@ func getHeaders(route *plugins.AppPluginRoute, orgId int64, appID string) (http.
...
+	data := templateData{
+		JsonData:       query.Result.JsonData,
+		SecureJsonData: query.Result.SecureJsonData.Decrypt(),
+	}
+	interpolated, err := InterpolateString(route.Url, data)
+	if err != nil {
+		return err
+	}
+	// update the url of the route
+	route.Url = interpolated
```
Comment (torkelo) [inline pkg/api/pluginproxy/pluginproxy.go]: Wont this modify the route url permanently for all, is this instance of a AppPluginRoute not the source defintiion object?

Comment (briangann) [reply pkg/api/pluginproxy/pluginproxy.go]: yes, i've fixed it and tested now, we're closing to be able to merge this (will remove the debug/trace info)

**Review by marefr — APPROVED** [review]
LGTM

**Review by marefr — CHANGES_REQUESTED** [review]
After consideration I think some information about this feature should be added to the [app plugin documentation](https://grafana.com/docs/plugins/developing/apps/). I can see that there are a specific page [Authentication for datasource plugins](https://grafana.com/docs/plugins/developing/auth-for-datasources/#plugin-routes) which currently holds the only documentation about datasource plugin routes. Since this feature is only about dynamic urls I think it's fine to put this documentation in the Developing app plugins page.

I also created an issue to track the feature this PR solves.

**Review by marefr — APPROVED** [review]
Great :+1:

**Pull request merged**